trin-core/src/jsonrpc/handlers.rs

```diff
@@ -33,61 +36,97 @@ impl JsonRpcHandler {
                         Ok(self.discovery.write_with_warn().await.routing_table_info())
                     }
                 },
-                TrinEndpoint::HistoryEndpoint(endpoint) => match self.history_jsonrpc_tx.as_ref() {
-                    Some(tx) => proxy_query_to_history_subnet(tx, endpoint).await,
-                    None => Err("Chain history subnetwork unavailable.".to_string()),
-                },
-                TrinEndpoint::StateEndpoint(endpoint) => match self.state_jsonrpc_tx.as_ref() {
-                    Some(tx) => proxy_query_to_state_subnet(tx, endpoint).await,
-                    None => Err("State subnetwork unavailable.".to_string()),
-                },
-                _ => Err(format!(
+                TrinEndpoint::OverlayEndpoint(endpoint) => {
+                    self.process_overlay_request(endpoint, request.params.clone())
+                        .await
+                }
+                _ => Err(JsonRpcError::InvalidRequest(format!(
                     "Can't process portal network endpoint {:?}",
                     request.endpoint
-                )),
+                ))),
+            };
+            let _ = match response {
+                Ok(val) => request.resp.send(Ok(val)),
+                Err(msg) => request.resp.send(Err(msg.to_string())),
             };
-            let _ = request.resp.send(response);
+        }
+    }
+
+    async fn process_overlay_request(
+        &self,
+        endpoint: OverlayEndpoint,
+        params: Params,
+    ) -> Result<Value, JsonRpcError> {
+        match params.clone() {
+            Params::Array(val) => match &val[0].as_str() {
+                Some("history") => match self.history_jsonrpc_tx.as_ref() {
+                    Some(tx) => proxy_query_to_history_subnet(tx, endpoint, params.clone()).await,
+                    None => Err(JsonRpcError::UnavailableNetwork("history".to_string())),
+                },
+                Some("state") => match self.state_jsonrpc_tx.as_ref() {
+                    Some(tx) => proxy_query_to_state_subnet(tx, endpoint, params).await,
```

Comment:
Do you think it is a good idea instead of passing the same `Params` enum (with the network identifier) to try to exclude the first value (the network identifier) before sending it to the overlay handlers? I think this may introduce some redundancy later in the data flow when we will be interested only in the actual *params.

I'm thinking about two ways to accomplish this:

- Add additional `OverlayParams` enum and a `try_from` trait for converting from `Params`. Then pass `OverlayParams` to the overlay handlers.
- Use the same `Params` enum but implement an additional method ("to_overlay_params" or something similar) which will remove the first network identifier element.